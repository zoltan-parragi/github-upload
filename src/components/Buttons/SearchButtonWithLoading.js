import React from "react";

const SearchButton = ({ className = "", children}) => (
	<button className={className} type="submit">
		{children}
	</button>
);

const Loading = () => 
    <div className="button-search">Loading... </div> 

const withLoading = (Component) => ({isLoading, ...rest}) =>
    isLoading
        ? <Loading />
        : <Component {...rest} />

const SearchButtonWithLoading = withLoading(SearchButton);

export default SearchButtonWithLoading;
export {
	SearchButtonWithLoading,
	SearchButton,
	Loading,
	withLoading
};