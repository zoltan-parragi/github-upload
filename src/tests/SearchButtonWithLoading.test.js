import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import SearchButtonWithLoading, {Loading, SearchButton} from '../components/Buttons';

describe('SearchButton', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<SearchButton>Search button</SearchButton>, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	test('has a valid snapshot', () => {
		const component = renderer.create(
			<SearchButton>Search button</SearchButton>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
})

describe('Loading', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<Loading />, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	test('has a valid snapshot', () => {
		const component = renderer.create(
			<Loading />
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});

describe('SearchButtonWithLoading', () => {
	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<SearchButtonWithLoading>Search button</SearchButtonWithLoading>, div);
		ReactDOM.unmountComponentAtNode(div);
	});
	test('has a valid snapshot', () => {
		const component = renderer.create(
			<SearchButtonWithLoading>Search button</SearchButtonWithLoading>
		);
		const tree = component.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
