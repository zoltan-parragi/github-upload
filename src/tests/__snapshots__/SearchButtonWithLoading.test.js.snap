// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading has a valid snapshot 1`] = `
<div
  className="button-search"
>
  Loading... 
</div>
`;

exports[`SearchButton has a valid snapshot 1`] = `
<button
  className=""
  type="submit"
>
  Search button
</button>
`;

exports[`SearchButtonWithLoading has a valid snapshot 1`] = `
<button
  className=""
  type="button"
>
  Search button
</button>
`;
